### ISSUE TITLE
Adding a class ".visible-XX-inline-table" or changing ".visible-*" entirely

### ISSUE THREAD

**Issue by MarcoGiancarli** [twbs/bootstrap#16304]
Currently, it is impossible to use the `.visible-XX` classes along with input groups. Input groups require the _display_ property to be set to _inline-table_. Currently, `.visible-*` classes are only capable of using the values

 * _block_
 * _inline-block_
 * _inline_
 * _table_
 * _table-row_
 * _table-cell_

Because of this, it is impossible to have a working element with both the classes `.visible-*` and `input-group`. Currently, as a workaround, the desired behavior can be achieved using three `.hidden-*` classes in place of one `.visible-*` class, but it would be more readable to have a `.visible-XX-inline-table` class. [Here is an example](http://jsfiddle.net/MarcoGiancarli/3teneete/5/) of how one of these new classes could be used with an input group. Resize your browser to test the changes in behavior.

Alternatively, a more robust solution would be to change the `.visible-*` classes such that they did not override the _display_ property. Essentially, the `.visible-xs` class would set _display_ to _none_ for sm/md/lg screen sizes, and change __nothing else__ (and likewise for the other three classes). This shouldn't break any correct code, but it would render `.visible-XX-*` classes unnecessary.

I'd be happy to make a pull request once it's clear what would be a better decision for handling this bug.

**Comment by twbs-lmvtfy** [comment]
Hi @MarcoGiancarli!

You appear to have posted a live example (http://fiddle.jshell.net/MarcoGiancarli/3teneete/4/show/light/), which is always a good first step. However, according to [the HTML5 validator](http://html5.validator.nu), **your example has some validation errors**, which might potentially be causing your issue:
* line 41, column 3 thru line 41, column 8: Start tag ```` body ```` seen but an element of the same type was already open.
* line 109, column 9 thru line 109, column 38: Duplicate ID ```` form2 ````.
* line 134, column 1 thru line 134, column 7: Saw an end tag after ```` body ```` had been closed.

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by MarcoGiancarli** [comment]
I revised the example and fixed the link in my original post.

**Comment by cvrebert** [comment]
> Adding a class ".visible-XX-inline-table"

I'm going to call that an approximate duplicate of #14662. FYI @mdo.

> Alternatively, a more robust solution would be to change the `.visible-*` classes such that they did not override the `display` property.

I'm pleased to say that we've already addressed this problem in our internal draft of Bootstrap v4 by only having `hidden-*` classes (which have been revamped and can be combined so as to have the same effect as the old `.visible-*` classes, except without the `display` override problem).
We're trying to minimize changes to Bootstrap v3 at this point (so as to focus on v4), so I believe we're going to leave this as "Fixed in v4, CantFix/WontFix in v3".
